internal/common/compose_result_test.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestWriteFull(t *testing.T) {
+
+	const testOptions = `{"msg": "test"}`
+
+	testStage := Stages{{
+		Name:    "testStage",
+		Options: []byte(testOptions),
+		Success: true,
+		Output:  "Finished",
+	}}
+
+	testBuild := Build{
+		Stages:  testStage,
+		TreeID:  "treeID",
+		Success: true,
+	}
+
+	testAssembler := Assembler{
+		Name:    "testAssembler",
+		Options: []byte(testOptions),
+		Success: true,
+		Output:  "Done",
+	}
+
+	testComposeResult := ComposeResult{
+		TreeID:    "TreeID",
+		OutputID:  "OutputID",
+		Build:     &testBuild,
+		Stages:    testStage,
+		Assembler: &testAssembler,
+		Success:   true,
+	}
+
+	var b bytes.Buffer
+	assert.NoError(t, testComposeResult.Write(&b))
+	expectedMessage := "Build pipeline:\nStage testStage\n{\n  \"msg\": \"test\"\n}\n\nOutput:\nFinished\nStages:\nStage: testStage\n{\n  \"msg\": \"test\"\n}\n\nOutput:\nFinished\nAssembler testAssembler:\n{\n  \"msg\": \"test\"\n}\n\nOutput:\nDone\n"
```

Comment:
I suggest using the raw string literal here: https://stackoverflow.com/questions/7933460/how-do-you-write-multiline-strings-in-go/7933487

Reply:
I just copied what it gave me and used it for comparison. I was thinking that this will just watch out for potential changes to this in the future. But it does look better with the formatted string, thanks!

Reply:
I would do the same with copying. :)